Svc/Version/Version.cpp

```diff
@@ -0,0 +1,233 @@
+// ----------------------------------------------------------------------
+// functions to log tlm on versions
+void Version ::fwVersion_tlm() {
+    Fw::LogStringArg fw_event = (Project::Version::FRAMEWORK_VERSION);
+    this->log_ACTIVITY_LO_FrameworkVersion(fw_event);
+    Fw::TlmString fw_tlm = (Project::Version::FRAMEWORK_VERSION);
+    this->tlmWrite_FrameworkVersion(fw_tlm);
+}
+
+void Version ::projectVersion_tlm() {
+    Fw::LogStringArg proj_event = Project::Version::PROJECT_VERSION;
+    this->log_ACTIVITY_LO_ProjectVersion(proj_event);
+    Fw::TlmString proj_tlm = Project::Version::PROJECT_VERSION;
+    this->tlmWrite_ProjectVersion(proj_tlm);
+}
+
+void Version ::libraryVersion_tlm() {
+
+    // Process libraries array
+    for (U8 i = 0; i < Project::Version::LIBRARY_VERSIONS_COUNT; i++) {
+        // Emit Event/TLM on library versions
+        this->log_ACTIVITY_LO_LibraryVersions(Fw::LogStringArg(Project::Version::LIBRARY_VERSIONS[i]));
+        // Write to Events
+        switch (i) {
+            case VER_SLOT_00:
+                this->tlmWrite_LibraryVersion01(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_01:
+                this->tlmWrite_LibraryVersion02(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_02:
+                this->tlmWrite_LibraryVersion03(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_03:
+                this->tlmWrite_LibraryVersion04(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_04:
+                this->tlmWrite_LibraryVersion05(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_05:
+                this->tlmWrite_LibraryVersion06(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_06:
+                this->tlmWrite_LibraryVersion07(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_07:
+                this->tlmWrite_LibraryVersion08(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_08:
+                this->tlmWrite_LibraryVersion09(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            case VER_SLOT_09:
+                this->tlmWrite_LibraryVersion10(Fw::TlmString(Project::Version::LIBRARY_VERSIONS[i]));
+                break;
+            default:
+                // It is possible to have more than 10 library versions; however design agreed to only
+                //  provide 10 TLM channels for it
+                break;
+        }
+    }
+}
+
+void Version ::customVersion_tlm_all() {
+    for (U8 i = 0; (m_enable == true) && (m_num_custom_elements != 0) && (i < Svc::VersionCfg::VersionEnum::NUM_CONSTANTS);
```

Comment:
Reading some parts of the docs/code I was under the impression that `m_enable` would control whether to send both telemtry and events, or only events.
Currently it looks like nothing will be downlinked unless m_enable is true.
Should there be a condition that still does a minimal amount of downlinking (maybe send an event only but no telemetry?) if the user sends a command `VERSION(CUSTOM/ALL)` when m_enable is false? 

For example I can see that users would expect to get some amount of info back if they specifically send `VERSION(CUSTOM)`, regardless of the verbose status.

Reply:
That's a great point. One way the users can get custom version is to first send the enable command and then ask for `VERSION(CUSTOM)` but that seems like a cumbersome way of doing it. I think when a user sends a command to request custom version, it should be sent directly. @LeStarch: Thoughts?